db_access.py

```diff
@@ -24,16 +30,25 @@ def get_all_users():
     return conn.execute('SELECT id, handle, admin FROM user').fetchall()
 
 
+def user_for(uid):
+    conn = get_db()
+    for a in conn.execute('SELECT handle FROM user WHERE id=:id', {'id':uid}):
+        return a[0]
+    return ''
+
 def delete_user(user_id):
     conn = get_db()
     conn.execute('DELETE FROM user WHERE id = :id', {'id': user_id})
     conn.commit()
 
 
-def add_user(handle):
+def add_user(handle, password):
+    password = hash_ps(password)
     conn = get_db()
-    conn.execute('INSERT INTO user (handle, admin) values (:handle, :admin)',
-                 {'handle': handle, 'admin': 0})
+    for c in conn.execute('SELECT * FROM user WHERE handle=:handle', {'handle':handle}):
+        return list(c)
+    conn.execute('INSERT INTO user (handle, password, admin) values (:handle, :password, :admin)',
+                 {'handle': handle, 'password':password, 'admin': 0})
```

Comment:
Also, returning a list of users from an `add_user` method is not at all expected, we need a better way.

My suggestion: create a method `get_user_by_handle` that tries to retrieve a user. Then call `add_user` if no users are found, flash an error otherwise.